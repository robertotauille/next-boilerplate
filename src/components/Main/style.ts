import styled from 'styled-components';

export const Wrapper = styled.main`
  width: 100%;
  height: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`;

export const Title = styled.h1`
  font-size: 1.7rem;
`;

export const Description = styled.h2`
  font-size: 1.5rem;
  font-weight: 400;
`;
