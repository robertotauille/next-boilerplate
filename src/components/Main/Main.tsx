import * as S from './style';

const Main = () => (
  <S.Wrapper>
    <S.Title>Next.JS Boilerplate</S.Title>
    <S.Description>go go go robertinho!</S.Description>
  </S.Wrapper>
);

export default Main;
