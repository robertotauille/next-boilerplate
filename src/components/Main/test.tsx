import { render, screen } from '@testing-library/react';

import Main from './Main';

describe('<Main />', () => {
  it('Deve renderizar o titulo', () => {
    const { container } = render(<Main />);

    expect(
      screen.getByRole('heading', { name: /Next.JS Boilerplate/i })
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('Deve renderizar a descrição', () => {
    const { container } = render(<Main />);

    expect(
      screen.getByRole('heading', { name: /go go go robertinho!/i })
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
