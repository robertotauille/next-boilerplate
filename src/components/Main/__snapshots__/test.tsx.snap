// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> Deve renderizar a descrição 1`] = `
<main
  class="style__Wrapper-zegsms-0 kfjnWJ"
>
  <h1
    class="style__Title-zegsms-1 daelfe"
  >
    Next.JS Boilerplate
  </h1>
  <h2
    class="style__Description-zegsms-2 fiUoQm"
  >
    go go go robertinho!
  </h2>
</main>
`;

exports[`<Main /> Deve renderizar o titulo 1`] = `
<main
  class="style__Wrapper-zegsms-0 kfjnWJ"
>
  <h1
    class="style__Title-zegsms-1 daelfe"
  >
    Next.JS Boilerplate
  </h1>
  <h2
    class="style__Description-zegsms-2 fiUoQm"
  >
    go go go robertinho!
  </h2>
</main>
`;
